lnwallet/chancloser/rbf_coop_transitions.go

```diff
@@ -0,0 +1,1109 @@
+		// variant of the co-op close tx.
+		//
+		// TODO(roasbeef): db will only store one instance -- which is ok?
+		err = env.ChanObserver.MarkCoopBroadcasted(closeTx, true)
+		if err != nil {
+			return nil, err
+		}
+
+		broadcastEvent := protofsm.DaemonEventSet{&protofsm.BroadcastTxn{
+			Tx: closeTx,
+			Label: labels.MakeLabel(
+				labels.LabelTypeChannelClose, &env.Scid,
+			),
+		}}
+
+		transitionEvent := fn.NewLeft[LocalSigReceived, OfferReceivedEvent](*msg)
+
+		chancloserLog.Infof("ChannelPoint(%v): received sig from "+
+			"remote party, broadcasting: tx=%v", env.ChanPoint,
+			lnutils.SpewLogClosure(closeTx),
+		)
+
+		return &CloseStateTransition{
+			NextState: &ClosePending{
+				transitionEvents: transitionEvent,
+				CloseTx:          closeTx,
+			},
+			NewEvents: fn.Some(protofsm.EmittedEvent[ProtocolEvent]{
+				ExternalEvents: fn.Some(broadcastEvent),
+			}),
+		}, nil
+	}
+
+	return nil, fmt.Errorf("%w: received %T while in LocalOfferSent",
+		ErrInvalidStateTransition, event)
+}
+
+// ProcessEvent implements the state transition function for the
+// RemoteCloseStart. In this state, we'll wait for the remote party to send a
+// closing_complete message. Assuming they can pay for the fees, we'll sign it
+// ourselves, then transition to the next state of RemoteOfferReceived.
+func (l *RemoteCloseStart) ProcessEvent(event ProtocolEvent, env *Environment,
+) (*CloseStateTransition, error) {
+
+	switch msg := event.(type) {
+	// If we receive a OfferReceived event, we'll make sure they can
+	// actually pay for the fee. If so, then we'll counter sign and
+	// transition to a terminal state.
+	case *OfferReceivedEvent:
+		// To start, we'll perform some basic validation of the sig
+		// message they've sent.
+		switch {
+		// We'll validate that the remote party actually has enough
+		// fees to pay the closing fees.
+		case !l.RemoteCanPayFees(msg.SigMsg.FeeSatoshis):
+			return nil, fmt.Errorf("%w: %v vs %v",
+				ErrRemoteCannotPay,
+				msg.SigMsg.FeeSatoshis,
+				l.RemoteBalance.ToSatoshis())
+
+		// The sequence they send can't be the max sequence, as that would
+		// prevent RBF.
+		case msg.SigMsg.Sequence > mempool.MaxRBFSequence:
+			return nil, fmt.Errorf("%w: %v", ErrNonFinalSequence,
+				msg.SigMsg.Sequence)
+		}
+
+		// With the basic sanity checks out of the way, we'll now
+		// figure out which signature that we'll attempt to sign
+		// against.
+		var (
+			remoteSig input.Signature
+			noClosee  bool
+		)
+		switch {
+		// If our balance is dust, then we expect the CloserNoClosee
+		// sig to be set.
+		case l.LocalAmtIsDust():
+			if msg.SigMsg.CloserNoClosee.IsNone() {
+				return nil, ErrCloserNoClosee
+			}
+			msg.SigMsg.CloserNoClosee.WhenSomeV(func(s lnwire.Sig) {
+				remoteSig, _ = s.ToSignature()
+				noClosee = true
+			})
+
+		// Otherwise, we'll assume that CloseAndClosee is set.
+		//
+		// TODO(roasbeef): NoCloserClosee, but makes no sense?
```

Comment:
I think we should still write code to handle this case even though the economics are silly

Reply:
Sure, my efforts went into just trying to remove it from the spec. Feels like a premature optimization for something that's very rare in practice.

Reply:
Yeah agreed. Removing it from the spec would be nice.